.gitignore

```diff
@@ -19,3 +23,20 @@ pytest_session.txt
 .cache/
 .pytest_cache
 .coverage
+.pre-commit-config.yaml
+.clang-format
+.flake8
+.isort.cfg
+.mypy.ini
+.ruff.toml
+towncrier.toml
+
+# Jupyter Notebook
+*.ipynb
+
+# VsCode
+*.vscode
+.flake8
```

Comment:
Please remove these final 4 lines because they are already there in lines 28 to 31.